### ISSUE TITLE
Shutdown of the Core seems to be blocked by the MQTT server ([homeassistant.components.mqtt] No ACK from MQTT server in 10 seconds)

### ISSUE THREAD

**Issue by klacol** [home-assistant/addons#2250]
## The problem
When I restart the Core, it takes over three minutes to restart, because the shutdown of the Core seems to be blocked by the MQTT server.

## System Health

version | core-2021.10.7
-- | --
installation_type | Home Assistant OS
dev | false
hassio | true
docker | true
user | root
virtualenv | false
python_version | 3.9.7
os_name | Linux
os_version | 5.10.17-v8
arch | aarch64
timezone | Europe/Berlin

<details><summary>Home Assistant Cloud</summary>

logged_in | false
-- | --
can_reach_cert_server | ok
can_reach_cloud_auth | ok
can_reach_cloud | ok

</details>

<details><summary>Home Assistant Supervisor</summary>

host_os | Home Assistant OS 6.6
-- | --
update_channel | stable
supervisor_version | supervisor-2021.10.8
docker_version | 20.10.8
disk_total | 57.8 GB
disk_used | 26.7 GB
healthy | true
supported | true
board | rpi4-64
supervisor_api | ok
version_api | ok
installed_addons | Mosquitto broker (6.0.1), File editor (5.3.3), Samba share (9.5.1), Terminal & SSH (9.2.1), Duck DNS (1.14.0), JupyterLab (0.5.0), SQLite Web (3.1.0), InfluxDB (4.2.1), Home Assistant Google Drive Backup (0.105.2)

</details>

<details><summary>Lovelace</summary>

dashboards | 3
-- | --
resources | 1
error | /config/ui-lovelace.yaml not found
views | 15
mode | storage

</details>

- Add-on with the issue: Mosquitto broker
- Add-on release with the issue:  Current version: 6.0.1

## Problem-relevant configuration
I am not sure, If this is related, but I have configures a Bridge to an external MQTT Broker.

```yaml

```

## Traceback/Error logs
I see hundred of those meassages in the debug log:

```
2021-11-03 10:11:01 WARNING (MainThread) [homeassistant.core] Timed out waiting for shutdown stage 1 to complete, the shutdown will continue
...
2021-11-03 10:12:30 WARNING (MainThread) [homeassistant.components.mqtt] No ACK from MQTT server in 10 seconds (mid: 35730)
...
2021-11-03 10:12:01 WARNING (MainThread) [homeassistant.core] Timed out waiting for shutdown stage 2 to complete, the shutdown will continue
...

```

**Comment by Electronlibre2012** [comment]
Hello,

i have the same problem and it seems that the last update of Zigbee2mqtt make a mess in my config...leds lamp are less responsives and sometimes bug (stay on or off).

**Comment by clang13** [comment]
@klacol Did you ever find a solution to this problem? I'm experiencing it as well, with the exact same messages, hundreds of MQTT-related messages:
```
2022-01-09 12:58:34 WARNING (MainThread) [homeassistant.components.mqtt] No ACK from MQTT server in 10 seconds (mid: 6457)
2022-01-09 12:58:44 WARNING (MainThread) [homeassistant.components.mqtt] No ACK from MQTT server in 10 seconds (mid: 6458)
2022-01-09 12:58:54 WARNING (MainThread) [homeassistant.components.mqtt] No ACK from MQTT server in 10 seconds (mid: 6459)
2022-01-09 12:58:59 WARNING (MainThread) [homeassistant.core] Timed out waiting for shutdown stage 1 to complete, the shutd
own will continue
```
every time I restart through the UI. If I instead restart by killing the container, everything is fine, which makes sense, as it appears the delay is on shutdown rather than startup.

**Comment by klacol** [comment]
> @klacol Did you ever find a solution to this problem? I'm experiencing it as well, with the exact same messages, hundreds of MQTT-related messages:

unfortunately not, I am waiting patiently on every core restart :-(

> I disabled mqtt_statestream in my configuration and was able to restart as expected, with each restart taking 10-15 seconds. Re-enabling mqtt_statestream reverted to the behavior where restarts take several minutes

very good finding; the same here. So it is perhaps related to the statestream (MQTT bridge). I use Rabbit MQ in the version  3.8.2 without any TLS yet.

**Comment by Electronlibre2012** [comment]
hello,

for me it was a kodi integration that was the problem : the kodi player no longer works, so the integration block HA to restart.

You need to find which integration block the restart...

**Comment by Mariusthvdb** [comment]
> confirmed that this hang is related to sending mqtt_statestream-related messages on shutdown:

if you still experience this, you could/should post those findings in a related new issue to get some attention, this is closed and will get no more response.

I could add I see issues with my MQTT instance hanging too, and having issues every few restarts (not happening always) so Id be glad if this could be eliminated as source for that problem.
